src/PathParticle.ts

```diff
@@ -150,8 +150,13 @@ export class PathParticle extends Particle
 		if(lerp >= 0 && this.path)
 		{
 			//increase linear movement based on speed
-			const speed = this.speedList.interpolate(lerp) * this.speedMultiplier;
-			this.movement += speed * delta;
+			if(this._doSpeed) {
+				const speed = this.speedList.interpolate(lerp) * this.speedMultiplier;
+				this.movement += speed * delta;
+			} else {
+				const speed = ParticleUtils.length(this.velocity);
```

Comment:
Getting the length of the velocity is inefficient (`sqrt()` is involved) - I would go with `this.speedList.current.value * this.speedMultiplier`.

Reply:
Good point! Got it updated.